index.html

```diff
@@ -1812,8 +1812,80 @@ <h3>Verify Proof</h3>
       <section class="normative">
         <h3>Retrieve Verification Method</h3>
 
-      </section>
+        <p>
+The following algorithm specifies how to safely retrieve a verification method,
+such as a cryptographic public key, by using a <a>verification method</a>
+identifier contained in a <a>data integrity proof</a>. Required inputs are a
+<a>data integrity proof</a> (<var>proof</var>) and a set of
+<strong>dereferencing options</strong> (<var>options</var>). A
+<strong>verification method</strong> is produced as output.
+        </p>
 
+        <ol class="algorithm">
+          <li>
+Let <var>verificationMethod</var> be set to a `null` value.
+          </li>
+          <li>
+Let <var>vmIdentifier</var> be set to
+<var>proof</var>.<var>verificationMethod</var>.
+          </li>
+          <li>
+Let <var>vmPurpose</var> be set to <var>proof</var>.<var>proofPurpose</var>.
+          </li>
+          <li>
+If <var>vmIdentifier</var> is not a valid URL, an
+`INVALID_VERIFICATION_METHOD_URL` error MUST be raised.
+          </li>
+          <li>
+Let <var>controllerDocument</var> be the result of dereferencing
+<var>vmIdentifier</var>.
+          </li>
+          <li>
+If <var>controllerDocument</var> is not a valid <a>controller document</a>,
+an `INVALID_CONTROLLER_DOCUMENT` error MUST be raised.
+          </li>
+          <li>
+Let <var>verificationMethodArray</var> be an empty array. If the value of
+the <var>vmPurpose</var> property in the <var>controllerDocument</var> is a
```

Comment:
I think we will also need to talk about the acceptable contexts in the `controllerDocument` (or even a default context if no `@context` is present). We can turn this comment into an issue to handle it in another PR if we don't want to do it here.

Reply:
Yes, and an example would probably make this clearer.

Reply:
Implemented in 764e27e5c260e205f4a61da12e8c9d604cbafb05.